### DISCUSSION TITLE
Allow configuring the max upload size for files

### DISCUSSION THREAD

**Discussion by neumantm** [pretix/pretix#2188] [Ideas] [answered]
Probably relevant line: https://github.com/pretix/pretix/blob/7ebefa7b85ea8e61dd462005e67c435c893a4188/src/pretix/settings.py#L846

It is currently hard coded.
I would  like to set it from `pretix.cfg`.
The current hard coded value could be used as the default.

**Comment by raphaelm** [comment]
No, this is not the relevant line. [DATA_UPLOAD_MAX_MEMORY_SIZE](https://docs.djangoproject.com/en/3.2/ref/settings/#data-upload-max-memory-size) is used for e.g. form data, not for file uploads.

The 10 MB limit for file upload is in different places throughout the codebase, see e.g. this search:

https://github.com/pretix/pretix/search?q=max_size

Having this in pretix.cfg would be nice, agreed! This should be an easy PR to create :)

**Reply by neumantm** [reply]
I see.
I guess then it is probably wise to have multiple configuration options for the different kinds of files (`image`, `favicon`, `email-attachment`, `other`).
What do you think?

**Comment by neumantm** [comment, marked as answer]
Resolved by https://github.com/pretix/pretix/pull/2199

**Comment by neumantm** [comment]
I just noticed, that when using the docker container, the upload size is capped at 100M by the internal nginx.
Maybe we could also allow admins to overwrite that without having to overwrite the whole file.

**Reply by neumantm** [reply]
I think the easiest solution would be to move the `client_max_body_size` to a new file in `/etc/nginx/conf.d`.
Then admins could just overwrite this new file.

**Reply by neumantm** [reply]
Another solution would be to create a template and let the pretix.bash script instantiate that template with values from the environment variables.

**Reply by raphaelm** [reply]
So my first thought is "is this really the right place to store >100MB" and my second thought is "anyone with a non-trivial pretix deloyment at some point needs to modify their docker container anyways".

I'd be fine with a more include-heavy nginx config structure to make things easier to override. I don't think runtime templating of configuration files would be warranted.
